daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/api/DFDLPrimTypeEnum.java

```diff
@@ -14,16 +14,31 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+package org.apache.daffodil.runtime1.api;
 
-package org.apache.daffodil.runtime1.dpath
-
-import org.junit.Assert.assertTrue
-import org.junit.Test
-
-class TestNodeInfo {
-
-  @Test def test_nodeInfoCanBeConstructed(): Unit = {
-    assertTrue(NodeInfo.isXDerivedFromY("Byte", "Long"))
-  }
+/**
+ * An enumeration of all DFDL's simple types.
+ */
+public enum DFDLPrimTypeEnum {
```

Comment:
Is it standard to end a Java enum type with `Enum`? I'm not sure I've seen that before, but it's been a while since I've done Java. I don't think we have anything like `DFLDPrimType` that this could be confused with?

Reply:
Also, I think Java convention is to capitalize enum names? Should we do that too?

Reply:
I am disinclined to all-caps enum names that come from Java enum definitions. It will make them very ugly to use in Scala code. It's never been a convention I liked. Honestly, I think it's a holdover from C days of early computing when ALL CAPS NAMES were a way of shouting that you did get the point that you shouldn't hard-code magic numbers in your code. But it wasn't necessary even then when we didn't have fancy IDEs that would take you to the definition in one click. 

The suffix "Enum" was intended to make clear that "this is just an enum", but I think I will remove it, as Java enums are classes and can have methods and other richness, which we may want to add at some point.